devbuild.sh

```diff
@@ -1,76 +1,235 @@
 #!/bin/bash
-set -eu
-
-#cd to location of script
-MYDIR=$(cd "$(dirname "$(readlink -f -n "${BASH_SOURCE[0]}" )" )" && pwd -P)
 
+# usage instructions
 usage () {
-  echo "Usage: "
-  echo "  $0 PLATFORM COMPILER"
-  echo ""
-  echo "PLATFORM: Name of machine you are building on"
-  echo "COMPILER: (optional) compiler to use; valid options are 'intel', 'gnu'"
-  echo ""
-  echo "NOTE: This script is for internal developer use only;"
-  echo "See User's Guide for detailed build instructions"
+  printf "Usage: $0 [OPTIONS]...\n"
```

Comment:
You can actually group this all into one `printf` statement like this (and you won't need the \n's because they're part of your string):
```
printf "\
Usage: $0 [OPTIONS]...

OPTIONS
  -h, --help
...
NOTE: This script is for internal developer use only;
See User's Guide for detailed build instructions
"
```

Reply:
@gsketefian Would you prefer that I did this through directing a block into cat (see following code) or write one large printf (as suggested)?

```
cat << EOF
Usage: $0 [OPTIONS]...

OPTIONS
  -h, --help
      show this help guide
...
EOF
```

Reply:
I like to use printf, but up to you.

Reply:
I went with `cat << EOF` so that I can type the usage and setting messages WYSIWYG. I kept other printf statements for lines that I wanted to indent.